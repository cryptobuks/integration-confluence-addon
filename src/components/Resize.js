import React, {Component} from 'react'
import min from './resize-100.svg'
import max from './resize-full-screen.svg'
import './resize.css'
import Analytics from '../helpers/Analytics'

class Resize extends Component {
  constructor (props) {
    super(props)
    this.handleClick = this.handleClick.bind(this)
  }

  handleClick (evt) {
    evt.preventDefault()
    Analytics.track(`Confluence.Resize${this.props.size}`)
    this.props.action(evt)
  }

  render () {
    const icon = this.props.size === 'max' ? min : max
    return (
      <a
        className={`resize ${this.props.size}`}
        onClick={this.handleClick.bind(this)}>
        <span className="size">
          <img src={icon} width="20" height="20" alt="Resize" />
        </span>
      </a>
    )
  }
}

export default Resize
