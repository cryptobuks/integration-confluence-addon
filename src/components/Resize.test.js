import React from 'react'
import renderer from 'react-test-renderer'
import Resize from './Resize'

it('renders correctly', () => {
  const tree = renderer.create(
    <Resize size="max" action={() => {}} />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
