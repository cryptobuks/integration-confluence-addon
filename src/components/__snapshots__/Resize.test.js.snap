// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<a
  className="resize max"
  onClick={[Function]}
>
  <span
    className="size"
  >
    <img
      alt="Resize"
      height="20"
      src="resize-100.svg"
      width="20"
    />
  </span>
</a>
`;
